tools/src/main/scala/org/apache/spark/sql/rapids/tool/ui/QualificationReportGenerator.scala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids.tool.ui
+
+import java.io.{File, FileInputStream, FileOutputStream, PrintWriter}
+import java.text.SimpleDateFormat
+import java.util.Date
+
+import com.nvidia.spark.rapids.tool.qualification.Qualification
+import org.json4s.DefaultFormats
+import org.json4s.jackson.Serialization
+
+import org.apache.spark.SparkConf
+import org.apache.spark.internal.Logging
+import org.apache.spark.util.Utils.getPropertiesFromFile
+
+class QualificationReportGenerator(
+    conf: SparkConf,
+    provider: Qualification) {
+
+  import QualificationReportGenerator._
+
+  // TODO: use HDFS API to create the logoutput if necessary
```

Comment:
please file na issue with all the followups we know of.

If this doesn't work with hdfs, what does it do?  this might cause our nightly tests to fail, in which case need to turn off UI by default for now.